src/Compilers/CSharp/Portable/Symbols/DelegateCaches/FullyConcreteChecker.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// Checks whether a type or a method is fully constructed with fully concrete types as opposed to yet another generic parameter.
+    /// </summary>
+    internal class FullyConcreteChecker : CSharpSymbolVisitor<bool>
```

Comment:
>FullyConcreteChecker [](start = 19, length = 20)

This entire class appears to be functionally redundant with `TypeSymbolExtensions.ContainsTypeParameter`